### PR TITLE
Fix/multiple seats per pax

### LINKED ISSUE #726: PaxDetail - missing seat in sidebar
**Describe the bug**
In DEV, this passenger (https://dvl.wcogtas.org/gtas/paxDetail/691427/19264895) has no seat data displaying. The data is there in the PNR tab, but doesn't display . The issue may be a data issue because in the message, there are duplicate TIF segments for each passenger that may or may not be correct. The seat assignments for many of the dupes are either their actual seats or 'XXX'. Whether or not the data is valid, we probably need more robust handling here.

**Environment**
DEV

**Screenshots**
missingseatdata

### PR DESCRIPTION
**Pull request by tjwdev**
Fixes #726 

For now will select first seat amongst an array of seats returned. Other enhancements will be made to allow for multiple seats return and use.

### CODE REVIEW

File: src/pages/paxDetail/PaxDetail.js [hunk 1]
```diff
@@ -144,7 +147,7 @@ const PaxDetail = props => {
       gender: res.gender,
       nationality: res.nationality,
       residenceCountry: res.residenceCountry,
-      seat: res.seat,
+      seat: res.seats[0],
```
Comment (gtasautumn) [inline src/pages/paxDetail/PaxDetail.js:150]: We need to ensure the seats prop is an array before accessing it like one. If res.seats gets set to "" above, this code will bomb. You could use the ``asArray`` util in utils or something like it as a safety.

``seat: asArray(res.seats)[0]``

**Review by gtasautumn — CHANGES_REQUESTED** [review]
pls update to check that the seats prop is an array before assigning

**Comment by tjwdev** [comment]
Latest merge on back-end component and front-end component includes Apis prioritization enhancement as well as more fleshed out check and formatting for multiple seat returns for the side bar. 

This also includes an enhancement for the seat-chart to be able to accept more than one seat request on the side bar navigation.

File: src/pages/paxDetail/PaxDetail.js (hunk 1 above)
Comment (tjwdev) [reply src/pages/paxDetail/PaxDetail.js]: This was the beginning of the comma separated idea, this was the first pass through at the fix until it was mentioned about Apis priority. The latest commit now will both comma separate on the sidebar as well as enable the click to properly go through on to the seat chart.

File: src/pages/paxDetail/PaxDetail.js [hunk 2]
```diff
@@ -133,6 +133,19 @@ const PaxDetail = props => {
   };
 
   const paxinfoData = res => {
+    if(res.seats === null){
+      res.seats = "";
+    } else {
+      let sts = "";
+      res.seats.forEach((elem, idx) => {
```
Comment (gtasautumn) [inline src/pages/paxDetail/PaxDetail.js:140]: you can also use ``res.seats.join(",")`` here in place of the else clause, but this works

**Review by gtasautumn — COMMENTED** [review]
So this looks good, ran it against the backend pr. One question tho is what to do about messages with a null seat. The link to the seat page still gets generated but with no text. Not sure whether we should nix it in this case or put some default text there or what. @dje04001 ?

seatEmptyLink

**Comment by tjwdev** [comment]
Ah, I think perhaps putting N/A and disabling it as a link might work well? But removing the seat may be cleaner when it has none. I'm amenable to both directions.

**Comment by gtasautumn** [comment]
Just spoke to Dave, we can leave the link in but should put "N/A" in the seat text. Just need to make sure that doesn't cause a bug where the seat page can't find the n/a seat.

**Comment by tjwdev** [comment]
N/A should display when no seat present and work as a link now, had to futz with how we give passenger information state-wise to the seat-chart. Fixed a few other things I'd noticed that wouldn't work with a null OR multiple seats from a display perspective on the seat-chart passenger info section.

**Pull request merged**